turbo/snapshotsync/freezeblocks/block_snapshots.go

```diff
@@ -1118,6 +1118,28 @@ func noOverlaps(in []snaptype.FileInfo) (res []snaptype.FileInfo) {
 	return res
 }
 
+func SegmentsCaplin(dir string) (res []snaptype.FileInfo, missingSnapshots []Range, err error) {
```

Comment:
i dont like this repeated code with Segments... is there a way we make this generic, configurable, etc?